### ISSUE TITLE
with ESLint 1.4.x `eslint ./` will lint all files on a drive

### ISSUE THREAD

**Issue by jvdl** [eslint/eslint#3792]
With eslint 1.4.x a new globbing matcher was introduced and the `processPath()` method used to pass it an appropriate glob has introduced a bug whereby a relative path to the current directory in the form of `./` would end up with a glob of `/**/*.js` - in effect searching from the root of the drive.

To resolve this, it may be better, instead of replacing the `./` in `processPath()`, to use `path.normalize()`.

I've added a console.log to cli-engine.js at CLIEngine.prototype.executeOnFiles:

```js 
console.log('before processing pattern is:', patterns);
patterns = patterns.map(processPath(extensions));
console.log('after processing pattern is:', patterns);
```

With the current code:

```
[~/src/testrepo]$ ./node_modules/eslint/bin/eslint.js ./ ../ .
before processing pattern is: [ './', '../', '.' ]
after processing pattern is: [ '/**/*.js', '..//**/*.js', '**/*.js' ]
```

Note that the first pattern passed in, `./` is transformed to `/**/*.js`

With the `path.normalize()` fix:

```
[~/src/testrepo]$ ./node_modules/eslint/bin/eslint.js ./ ../ .
before processing pattern is: [ './', '../', '.' ]
after processing pattern is: [ '**/*.js', '../**/*.js', '**/*.js' ]
```

To summarise:

1. Run `eslint ./`
2. Expected result: JS files in the current directory are linted
3. Actual result: All JS files on the drive are linted
4. Suggested fix: use `path.normalize()` in favour of replacing `./` at the start of a path.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by rizowski** [comment]
I am experiencing this problem currently. 
When I pass eslint `./` it tries to lint `/Library/Application Support/Apple/ParentalControls/Users` then blows up because it doesn't have access. Ha I am in `/Users/username/git/eslint-watch` when trying to run the command.

**Comment by IanVS** [comment]
@BYK, can you take a look at this?

**Comment by nzakas** [comment]
@jvdl are you using 1.4.1?

**Comment by BYK** [comment]
Looking

**Comment by nzakas** [comment]
Working on this.

**Comment by jvdl** [comment]
@nzakas yes to clarify I was using 1.4.1
